### DISCUSSION TITLE
How to add nullable type when using MAP_VALUES_AS_ADDITIONAL_PROPERTIES?

### DISCUSSION THREAD

**Discussion by namgaxilem** [victools/jsonschema-generator#400] [Q&A] [answered]
Currently my config is like this:
![image](https://github.com/victools/jsonschema-generator/assets/43407988/fc14f4bf-93c8-49e5-af16-12bbd4ac8383)

And the generate chema is like this:
![image](https://github.com/victools/jsonschema-generator/assets/43407988/7f7a7836-d984-4366-8744-68e8093fd7da)

But I want my Map<String, String> can be nullable also, for example my json contain a null field, it will fail (for keyring-file):
![image](https://github.com/victools/jsonschema-generator/assets/43407988/c695f8e6-4555-4b4b-86a9-fd9d86d56016)

I tried withAdditionalPropertiesResolver but can not find a way to add null type belong with String type, my expectation is that type should be ["string", "null"] instead of "string" only.

**Comment by CarstenWickner** [comment, marked as answer]
Hi @namgaxilem,

For now, you might want to give this a try:
```java
configBuilder.forTypesInGeneral().withAdditionalPropertiesResolver((scope, context) -> {
        if (scope.getType().isInstanceOf(Map.class)) {
            ObjectNode schema = context.getGeneratorConfig().createObjectNode();
            ArrayNode anyOf = schema.putArray(context.getKeyword(SchemaKeyword.TAG_ANYOF));
            anyOf.addObject().put(context.getKeyword(SchemaKeyword.TAG_TYPE), context.getKeyword(SchemaKeyword.TAG_TYPE_NULL));
            anyOf.add(context.createStandardDefinitionReference(scope.getTypeParameterFor(Map.class, 1), null));
            return schema;
        }
        return null;
    });
```

Need to find some more time to think about how to best handle this case in a more standardized way.

**Reply by namgaxilem** [reply]
Thank you very much for your response, in my case, I applied your code for both forTypesInGeneral and forFields and it work!
